upgrade to 23.1 (#4)
DevExpress-Examples/maui-data-form: CS/ComboBoxEditor/Readme.md

<img src="../../Images/data-form-combobox-iphone.png" alt="DevExpress Data Form for iOS" height="700"/> <img src="../../Images/data-form-combobox-android.png" alt="DevExpress Data Form for Android" height="700"/>

### Configure Combo Box Settings

To specify an item source for the data form's combo box editor, follow the steps below:

> 1. Create a class that implements the [IPickerSourceProvider](https://docs.devexpress.com/Maui/DevExpress.Maui.DataForm.IPickerSourceProvider) interface. An instance of this class (*ComboBoxDataProvider*, in this example) serves as the data source for all combo box editors within the data form. 

Reviewer: ```suggestion
1. Create a class that implements the [IPickerSourceProvider](https://docs.devexpress.com/Maui/DevExpress.Maui.DataForm.IPickerSourceProvider) interface. An instance of this class (*ComboBoxDataProvider* in this example) serves as the data source for all combo box editors within the data form. 
```

In this example, the data form is bound to the *EmployeeInfo* object and uses combo box editors for the following properties of this object:
    - *Department*  
    The data provider's `GetSource` method returns a list of `DepartmentInfo` objects that serve as the data source for the drop-down item list of the **Department** combo box. Use the [ValueMember](https://docs.devexpress.com/MAUI/DevExpress.Maui.DataForm.DataFormComboBoxItem.ValueMember) and [DisplayMember](https://docs.devexpress.com/MAUI/DevExpress.Maui.DataForm.DataFormComboBoxItem.DisplayMember) parameters to specify data source field names that supply item values (integer values that define department codes) and display strings (department names), respectively.
    - *Status*  
    The data provider's `GetSource` method returns a list of strings that are displayed as items within the **Status** combo box and used to set the data object's *Status* property value.

### Specify Editor Navigation Settings

> When a user taps a [DataFormView](https://docs.devexpress.com/Maui/DevExpress.Maui.DataForm.DataFormView) editor, its state turns into Focused and the on-screen keyboard appears. The user can tap the **Next** button on the keyboard to move on to the next editor within the [DataFormView](https://docs.devexpress.com/Maui/DevExpress.Maui.DataForm.DataFormView). The [DataFormView](https://docs.devexpress.com/Maui/DevExpress.Maui.DataForm.DataFormView) shows the **Next** button for a focused editor if its [ReturnType](https://docs.devexpress.com/MAUI/DevExpress.Maui.DataForm.DataFormTextItemBase.ReturnType) property is set to `Next`. To show the keyboard for a [combo box editor](https://docs.devexpress.com/MAUI/DevExpress.Maui.DataForm.DataFormComboBoxItem), you should additionally set its [IsFilterEnabled](https://docs.devexpress.com/MAUI/DevExpress.Maui.DataForm.DataFormComboBoxItem.IsFilterEnabled) property to `True`.

Reviewer: ```suggestion
When a user taps a [DataFormView](https://docs.devexpress.com/Maui/DevExpress.Maui.DataForm.DataFormView) editor, its state changes to Focused and the on-screen keyboard appears. The user can tap the **Next** button on the keyboard to move to the next editor in the [DataFormView](https://docs.devexpress.com/Maui/DevExpress.Maui.DataForm.DataFormView). The [DataFormView](https://docs.devexpress.com/Maui/DevExpress.Maui.DataForm.DataFormView) shows the **Next** button for a focused editor if its [ReturnType](https://docs.devexpress.com/MAUI/DevExpress.Maui.DataForm.DataFormTextItemBase.ReturnType) property is set to `Next`. To show the keyboard for a [combo box editor](https://docs.devexpress.com/MAUI/DevExpress.Maui.DataForm.DataFormComboBoxItem), you should additionally set its [IsFilterEnabled](https://docs.devexpress.com/MAUI/DevExpress.Maui.DataForm.DataFormComboBoxItem.IsFilterEnabled) property to `True`.
```
